app/javascript/src/controller_branches.js

```diff
@@ -171,6 +177,26 @@ BranchesController.createBranch = function($node) {
   return branch;
 }
 
+BranchesController.addBranchCombinator = function(args) {
+  var branch = args[0];
+  if( !branch.index == 0 ) {
+    branch.$node.before("<p class=\"branch-or\">or</p>");
+  }
+}
+
+BranchesController.removeBranchCombinator = function(node) {
+    var $or = $(node).prev('.branch-or').first();
+    if($or) {
+      $or.remove()
+    }
+}
+
```

Comment:
Since we're dealing with ES6 (same as nodejs) environments rather than older JS approach (ie. each file has it's own scope rather than, by default, sharing the same window scope), there is no absolute requirement to create functions in the form:
```
 BranchesController.someSpecificControllerFunctionHere = function() {
   // blah
 }
```

This is a habit I developed with older JS code to indicate the function is a Class-level method (not instance) and not some random function (which, in older JS would have resulted in window.myFunctionExistsHere scope).

The intention was to show the explicit relationship with the inner workings of the Class (in this case the BranchController). 

In the current JS (ES6) environment, these two functions share the same scope, so you could argue the the Class name prefix version is not required:
```
 BranchesController.someSpecificControllerFunctionHere = function() {
   // blah
 }
 
 function addBranchEventListeners() {
   // blah
 }
```

As said though, the one added to the Class is trying to show it should not be extracted out to its own file (because sometimes things were thrown into the same file by error/time-saving/lack of judgement, etc.) and could be better.

Hopefully that explains the though-process behind the `BranchesController.something` approach. 

All that aside, however, there is this code inside the BranchController#create aciton:
```
BranchesController.enhanceCurrentBranches.call(this, $branches);
BranchesController.enhanceBranchInjectors.call(this, $injectors);
BranchesController.enhanceBranchOtherwise.call(this, $otherwise);

addBranchEventListeners(view)
```

To follow with principles of code consistency, it should probably be:
```
BranchesController.enhanceCurrentBranches.call(this, $branches);
BranchesController.enhanceBranchInjectors.call(this, $injectors);
BranchesController.enhanceBranchOtherwise.call(this, $otherwise);
BranchesController.addBranchEventListeners(this);
```
or change those added directly onto the class to the same non _BranchesController.something_ form.

Reply:
I think given that there is another class `BranchInjector` within the same file.  Keeping the `BranchesController.` format is more explicit.